### PR TITLE
chore(eslint): setup eslint-plugin-unicorn

### PR DESCRIPTION
**Pull request by jean-smaug**
Rules available here https://github.com/sindresorhus/eslint-plugin-unicorn

### CODE REVIEW

File: packages/frontend/app/routes/reset-password.tsx [hunk 1]
```diff
@@ -90,13 +90,8 @@ export default function ResetPasswordPage() {
...
 
-  let pageTitle: string;
-
-  if (pageType === "s") {
-    pageTitle = "Set your password";
-  } else {
-    pageTitle = "Reset password";
-  }
+  const pageTitle: string =
+    pageType === "s" ? "Set your password" : "Reset password";
```
Comment (jean-smaug) [inline packages/frontend/app/routes/reset-password.tsx]: Code optim detection

File: packages/frontend/app/routes/reset-password.tsx [hunk 2]
```diff
@@ -68,11 +68,11 @@ export const action: ActionFunction = async ({ request }): Promise<ActionData> =
     await resetPassword(password!, token);
 
     return { success: true };
-  } catch (err) {
-    if (err instanceof Error) {
+  } catch (error) {
+    if (error instanceof Error) {
```
Comment (jean-smaug) [inline packages/frontend/app/routes/reset-password.tsx:72]: Prefere `error` over `err` or `e`

File: packages/frontend/app/routes/dashboard/projects/$id/settings.tsx [hunk 3]
```diff
@@ -50,13 +50,13 @@ export const action: ActionFunction = async ({ request, params }): Promise<Actio
 
   const promiseOfMembersToRemove: Array<Promise<{ statusCode: number }>> = [];
 
-  formData.forEach((d) => {
+  for (const d of formData) {
```
Comment (jean-smaug) [inline packages/frontend/app/routes/dashboard/projects/$id/settings.tsx:53]: prefere for...of over forEach

File: packages/frontend/app/routes/dashboard/projects/$id/environments/$env/flags/$flagId/insights.tsx [hunk 4]
```diff
@@ -141,7 +141,7 @@ const InsightsGrid = styled("div", {
 });
 
 const formatDefaultDate = (isoDate: string) => {
-  return isoDate.substr(0, 10);
+  return isoDate.slice(0, 10);
```
Comment (jean-smaug) [inline packages/frontend/app/routes/dashboard/projects/$id/environments/$env/flags/$flagId/insights.tsx:144]: prefere slice over `substr` or `substring`

File: packages/frontend/app/modules/forms/utils/validateEmail.ts [hunk 5]
```diff
@@ -3,7 +3,7 @@ export const validateEmail = (email?: string): string | undefined => {
     return "The email field is required.";
   }
 
-  if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(email)) {
+  if (!/^[\w%+.-]+@[\d.a-z-]+\.[a-z]{2,}$/i.test(email)) {
```
Comment (jean-smaug) [inline packages/frontend/app/modules/forms/utils/validateEmail.ts:6]: Regex validation and simplification.
E.g : `\d` instead of `[0-9]`

**Comment by jean-smaug** [comment]
@mfrachet This one is ready for the linting part.
Cleanup seems to fail randomly, but I don't see how my PR could make it fail 😬 

https://github.com/progressively-crew/progressively/actions/runs/3118011781/jobs/5057033463#step:13:44

**Comment by mfrachet** [comment]
I got inconsistencies form some reasons that I can't figure out sometimes on CRA and Nextjs due to the seeding. 

Thanks for the contrib, I merge :D

**Pull request merged**